File: app/src/main/java/com/iamagamedev/mybrandnewgame/SoundManager.java
package com.iamagamedev.mybrandnewgame;

import android.content.res.AssetFileDescriptor;
import android.content.res.AssetManager;
import android.media.AudioManager;
import android.media.SoundPool;

import java.io.IOException;

/**
 * Created by Михан on 24.05.2017.
 */

public class SoundManager {

    private SoundPool soundPool;
    private int fireBall = -1;
    private int hiroEnemy = -1;
    private int fireballExplode = -1;
    private static SoundManager soundManager;

    private SoundManager() {
    }

    public static SoundManager getInstance() {
        if (soundManager == null) {
            soundManager = new SoundManager();
        }
        return soundManager;
    }

    public void loadSound() {
        soundPool = new SoundPool(10, AudioManager.STREAM_MUSIC, 0);
        try {
            AssetManager assetManager = ThisApp.getInstance().getAssets();
            AssetFileDescriptor descriptor;

            descriptor = assetManager.openFd("PutThatCookieDown.ogg");
            fireBall = soundPool.load(descriptor, 0);

            descriptor = assetManager.openFd("hit_guard.ogg");
            hiroEnemy = soundPool.load(descriptor, 0);

            descriptor = assetManager.openFd("explode.ogg");
            fireballExplode = soundPool.load(descriptor, 0);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void playSound(String sound) {
        switch (sound) {
            case "PutThatCookieDown":
                soundPool.play(fireBall, 1, 1, 0, 0, 1);
                break;
            case "hit_guard":
                soundPool.play(hiroEnemy, 1, 1, 0, 0, 1);
                break;
            case "explode":
                soundPool.play(fireballExplode, 1, 1, 0, 0, 1);
                break;
        }
    }
    public void playPTCD(){
        soundPool.play(fireBall, 1, 1, 0, 0, 1);
    }
}

File: app/src/main/java/com/iamagamedev/mybrandnewgame/SoundManager.kt
package com.iamagamedev.mybrandnewgame

import android.content.res.AssetFileDescriptor
import android.media.AudioManager
import android.media.SoundPool

/**
 * Created by Михан on 24.05.2017.
 */
class SoundManager private constructor() {
    private var soundPool: SoundPool? = null
    private var fireBall = -1
    private var heroEnemy = -1
    private var fireballExplode = -1
    fun loadSound() {
        soundPool = SoundPool(10, AudioManager.STREAM_MUSIC, 0)
        try {
            val assetManager = ThisApp.getInstance().assets
            var descriptor: AssetFileDescriptor
            descriptor = assetManager.openFd("PutThatCookieDown.ogg")
            fireBall = soundPool!!.load(descriptor, 0)
            descriptor = assetManager.openFd("hit_guard.ogg")
            heroEnemy = soundPool!!.load(descriptor, 0)
            descriptor = assetManager.openFd("explode.ogg")
            fireballExplode = soundPool!!.load(descriptor, 0)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun playSound(sound: String?) {
        when (sound) {
            "PutThatCookieDown" -> soundPool!!.play(fireBall, 1f, 1f, 0, 0, 1f)
            "hit_guard" -> soundPool!!.play(heroEnemy, 1f, 1f, 0, 0, 1f)
            "explode" -> soundPool!!.play(fireballExplode, 1f, 1f, 0, 0, 1f)
        }
    }

    fun playPTCD() {
        soundPool!!.play(fireBall, 1f, 1f, 0, 0, 1f)
    }

    companion object {
        private var soundManager: SoundManager? = null
        val instance: SoundManager?
            get() {
                if (soundManager == null) {
                    soundManager = SoundManager()
                }
                return soundManager
            }
    }
}